### PR TITLE
Added contribution section in Readme

### LINKED ISSUE #2638: Add CONTRIBUTING.md File to this Repo
Please fill in the fields below to submit an issue or feature request.  The
more information that is provided, the better.

**Description of issue or feature request**:
The repository currently lacks a CONTRIBUTING.md file. This file is essential as it guides new contributors on how to get started with the project, outlines the code of conduct, provides guidelines for submitting issues and pull requests, and includes other relevant information for contributors.

**Current behavior**:
There is no CONTRIBUTING.md file present in the repository, which might lead to confusion for new contributors regarding the contribution process and guidelines.

**Expected behavior**:
A CONTRIBUTING.md file should be added to the root of the repository. This file should include:
- Instructions on how to set up the development environment.
- Guidelines for submitting issues and pull requests.
- Code of conduct.
- Style guides for coding, commits, and documentation.
- Any other relevant information for contributors.

This will help streamline the contribution process and ensure that all contributors

### PR DESCRIPTION
**Pull request by Ayush9026**
**Description of the changes being introduced by the pull request**:

Added Contributing and Community section in Readme along with [Developer documentation](https://theupdateframework.readthedocs.io/), including [API reference](https://theupdateframework.readthedocs.io/en/latest/api/api-reference.html) and [instructions for contributors](https://theupdateframework.readthedocs.io/en/latest/CONTRIBUTING.html)

Fixes https://github.com/theupdateframework/python-tuf/issues/2638

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,16 @@ for the reference implementation
 * [Miscellaneous Docs](https://github.com/theupdateframework/python-tuf/tree/develop/docs)
 * [Python-TUF development blog](https://theupdateframework.github.io/python-tuf/)
 
+Contribution and Community
+--------
+We welcome Contributions, updates, [discrepancy reports](https://github.com/theupdateframework/python-tuf/issues) and [pull requests](https://github.com/theupdateframework/python-tuf/pulls) are welcome. This project thrives on community collaboration. Contributors are expected to adhere to our [Code of Conduct](https://github.com/theupdateframework/python-tuf/blob/develop/docs/CODE-OF-CONDUCT.md).
+
+Jump into our [Slack](https://communityinviter.com/apps/cloud-native/cncf)! Our projects are community-built and welcome collaboration.
```
Comment (h4l0gen) [inline README.md:70]: Hi @Ayush9026 Hope you are doing well. IMO we don't need any contact details here (like Slack link). Just below that we have `Contact` section, which is the perfect place to define that and it is [already ](https://github.com/theupdateframework/python-tuf?tab=readme-ov-file#contact)there. WYT? 


Also as i said [here](https://github.com/theupdateframework/theupdateframework.io/pull/58#discussion_r1614667316), Please wait for official maintainer's review what they think, before making any changes. I hope you understand. Thank you😊👍

Comment (Ayush9026) [reply README.md:70]: Hi @h4l0gen,

Any PR doesn't get merged without an official maintainer's review, and I always mention the issue in the PR. I raised this PR with the permission of @lukpueh sir. I just wanted to thank you for your feedback and didn’t make any changes based on it yet. 
I understand the importance of patience in open source contributions. I have been contributing to different projects for some time, and I assure you that I follow the protocol of seeking maintainers' approval before raising any PR. Thank you for your understanding and support.

**Comment by jku** [comment]
On a first read this feels rendundant since it mostly repeats links that already exist elsewhere in the same README.

I don't mind re-organizing the README (I'm sure it could be better) but just adding the same links again in another paragraph feels bad for maintanability.

**Comment by lukpueh** [comment]
I agree with @jku.  

@Ayush9026, I thought we were on the same page regarding the contents of this patch. See https://github.com/theupdateframework/python-tuf/issues/2638#issuecomment-2126739469

**Comment by lukpueh** [comment]
> I agree with @jku.
> 
> @Ayush9026, I thought we were on the same page regarding the contents of this patch. See [#2638 (comment)](https://github.com/theupdateframework/python-tuf/issues/2638#issuecomment-2126739469)

... just saw that this already happened in https://github.com/theupdateframework/python-tuf/pull/2641. It think that's enough for now.

**Comment by Ayush9026** [comment]
Hi @jku , @lukpueh 

Thanks for the feedback. I understand the concerns about redundancy and maintainability. My intention was to make the guidelines more accessible.. I see now that repeating links already present in the README might not be the best approach.

**Pull request closed without merging**